cmake/Utils.cmake

```diff
@@ -216,3 +216,33 @@ function(pycmd outvar cmd)
   string(STRIP "${_output}" _output)
   set(${outvar} "${_output}" PARENT_SCOPE)
 endfunction()
+
+###
+# Helper function to print out everything that cmake knows about a target
+#
+# Copied from https://stackoverflow.com/questions/32183975/how-to-print-all-the-properties-of-a-target-in-cmake
+# This isn't called anywhere, but it's very useful when debugging cmake
+# NOTE: This doesn't work for INTERFACE_LIBRARY or INTERFACE_LINK_LIBRARY targets
+
+# Get all propreties that cmake supports
+execute_process(COMMAND cmake --help-property-list OUTPUT_VARIABLE CMAKE_PROPERTY_LIST)
```

Comment:
Likely all platforms will be fine with this, but, to be safe, I think we should wrap this in a function and cache the result to be used in the `print_target_properties` function below. For example

```
function(populate_cmake_property_list)
    if (!${CMAKE_PROPERTY_LIST})
        execute_process(COMMAND cmake --help-property-list OUTPUT_VARIABLE CMAKE_PROPERTY_LIST)
        ...
    endif()
endfunction()

function(print_target_properties tgt)
    populate_cmake_property_list()
    ... the rest of the function ...
endfunction(print_target_properties)
```

Thoughts?